### PR TITLE
[IN PROGRESS] Remove code-based authentication, and implement key-vault

### PR DESCRIPTION
**Pull request by bryanar**
- [x] Add Database Connection String to Azure Vault
- [x] Current issue I am working through : failing to retrieve token with our current certificate.
- [x] Test Cycle
- [x] Clean up

### CODE REVIEW

File: src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+namespace dumplingWeb.App_Start
+{
+    public class AuthConfig
+    {
+
+        public static class CertificateHelper
+        {
+            public static X509Certificate2 FindCertificateByThumbprint(string findValue)
+            {
+                X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine);
```
Comment (lt72) [inline src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs]: it seems like the X509Store class is IDisposable (from MSDN: public sealed class X509Store : IDisposable, https://msdn.microsoft.com/en-us/library/system.security.cryptography.x509certificates.x509store(v=vs.110).aspx) 

so this should be better coded (more clearly) as 

using(X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine))
{
/// ...
}

File: src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    return col[0];
+                }
+                finally
+                {
+                    store.Close();
+                }
+            }
+        }
+
+        private static AuthenticationContext _context;
```
Comment (lt72) [inline src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs]: do we really need _context to be a fieldof this class? it seems the only purpose it serves is to call AcquireTokenAsyn, for which a local variable in GetAccessToken method should be enough...

File: src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+            return result.AccessToken;
+        }
+
+        private static ClientAssertionCertificate _cert;
+        public static ClientAssertionCertificate AssertionCert
+        {
+            get
+            {
+                if (_cert == null)
```
Comment (lt72) [inline src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs]: here you use _cert directly... see comment below

File: src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                {
+                    var clientAssertionCertPfx = CertificateHelper.FindCertificateByThumbprint(WebConfigurationManager.AppSettings["thumbprint"]);
+                    AssertionCert = new ClientAssertionCertificate(WebConfigurationManager.AppSettings["dumpling_ad_client_id"], clientAssertionCertPfx);
```
Comment (lt72) [inline src/dumpling-service/dumpling.web/App_Start/AuthConfig.cs]: ... (see comment above)  here you do not.  if you always use _cert, or always use AssertionCert, then code will be more readable.  Using AssertionCert should allow you to get rid of _cert altogether I believe.

**Comment by bryanar** [comment]
@lt72  -  Likes mean I thought about it and I accept the change.

**Comment by bryanar** [comment]
oops

**Pull request closed without merging**